Userland/Services/AudioServer/Mixer.cpp

```diff
@@ -65,12 +68,18 @@ void Mixer::mix()
     decltype(m_pending_mixing) active_mix_queues;
 
     for (;;) {
-        if (active_mix_queues.is_empty() || m_added_queue) {
-            pthread_mutex_lock(&m_pending_mutex);
-            pthread_cond_wait(&m_pending_cond, &m_pending_mutex);
+        // While we have nothing to mix, wait on the condition.
+        if (m_pending_mixing.is_empty() && active_mix_queues.is_empty()) {
```

Comment:
This is the UB: you can't access `m_pending_mixing ` while not holding `m_pending_mutex `.

The pattern is, once again:

1. Lock the mutex.
2. Do the checks (while holding the mutex!).
3. If nothing to do (empty queue), wait on condition variable, then goto 2.
4. Act.
5. Release the mutex and go your merry way _OR_ (if in a loop) goto 2 without releasing the mutex.

In code:

```c++
mutex.lock();
condvar.wait([&] {
    return !queue.is_empty();
});
do_stuff(queue);
mutex.unlock();
```

Or better, don't hold the mutex while "doing stuff":

```c++
mutex.lock();
condvar.wait([&] {
    return !queue.is_empty();
});
auto stuff = dequeue_stuff(queue);
mutex.unlock();
do_stuff(stuff);
```